volk: accurate exp kernel. rebased #12

jdemel opened the pull request:
A more accurate exp VOLK kernel than volk_32f_expfast_32f. Taken from code licensed with zlib.

This is a continuation of #12 . I just rebased this PR onto current master and added the GPL header.

trondeau committed:
volk: accurate exp kernel.

A more accurate exp VOLK kernel than volk_32f_expfast_32f.Taken from
code licensed with zlib.

rear1019 commented on kernels/volk/volk_32f_exp_32f.h:
```diff
@@ -0,0 +1,298 @@
+#define INCLUDED_volk_32f_exp_32f_a_H
+
+#ifdef LV_HAVE_SSE4_1
+#include <smmintrin.h>
```
The code is actually SSE2. I don’t see any non-SSE2 instructions and just compiled using ``LV_HAVE_SSE2`` and including ``<emmintrin.h>`` instead of ``<smmintrin.h>``.

jdemel replied:
That's a surprise, since I just resubmitted this PR and we already marked it as reviewed. Hmm...

rear1019 reviewed (changes requested):
The code should be safe to compile with SSE2 instead of requiring SSE4.1.

jdemel committed:
exp: Rename SSE4.1 to SSE2 kernel

The SSE kernel only requires SSE2 instructions. Thus, we can just use
this instruction level.

jdemel commented:
@michaelld since this PR was already reviewed and is just a rebase. I'd say it is ready to be merged.

michaelld reviewed (approved):
LGTM
